src/signal_handler.rs

```diff
@@ -0,0 +1,154 @@
+use super::*;
+
+use command_group::{CommandGroup, GroupChild};
+use signal_hook::consts::signal::*;
+use signal_hook::consts::TERM_SIGNALS;
+use signal_hook::flag;
+use signal_hook::iterator::exfiltrator::WithOrigin;
+use signal_hook::iterator::SignalsInfo;
+use std::io::Read;
+use std::process::Command;
+use std::sync::{atomic::AtomicBool, Arc};
+
+pub(crate) struct SignalHandler {
+  verbosity: Verbosity,
+  signals: Option<SignalsInfo<WithOrigin>>,
+}
+
+impl SignalHandler {
+  pub(crate) fn install(verbosity: Verbosity) -> Result<(), std::io::Error> {
+    let mut instance = Self::instance();
+    instance.verbosity = verbosity;
+
+    let signal_flag = Arc::new(AtomicBool::new(false));
+
+    const MISC_SIGNALS: &[i32] = &[
+      #[cfg(not(windows))]
+      SIGHUP,
+      // SIGABRT,
+      // SIGPIPE,
+      // SIGALRM,
+    ];
+
+    let mut sigs = MISC_SIGNALS.to_vec();
+    sigs.extend(TERM_SIGNALS);
+
+    let signals = SignalsInfo::<WithOrigin>::new(&sigs)?;
+    instance.signals = Some(signals);
+
+    for sig in TERM_SIGNALS {
+      // NOTE This could be set before the register handler, in order to terminate in a double ctrl-c
+      // flag::register_conditional_shutdown(*sig, 1, Arc::clone(&signal_flag))?;
+      flag::register(*sig, Arc::clone(&signal_flag))?;
+    }
+
+    for sig in MISC_SIGNALS {
```

Comment:
```suggestion
    for signal in MISC_SIGNALS {
```

This loop looks like it's doing the same thing as the one above, can we combine them?